packages/ui-components/src/features/HookForm/HookForm.tsx

```diff
@@ -0,0 +1,28 @@
+/*
+ * Tupaia
+ *  Copyright (c) 2017 - 2023 Beyond Essential Systems Pty Ltd
+ */
+
+import React from 'react';
+import { FormProvider, SubmitHandler } from 'react-hook-form';
+
+interface FormProps {
+  formContext: any;
+  onSubmit?: SubmitHandler<any>;
+  children: React.ReactNode;
+  className?: string;
+}
+
+/**
+ * A wrapper around react-hook-form's FormProvider and form element. Needed for custom form fields
+ * such as TextField and CheckboxField to be able to self register and validate with react-hook-form.
+ */
+export const HookForm = ({ formContext, onSubmit, children, className }: FormProps) => {
```

Comment:
I don't feel too strongly so feel free to push back but it would be nice to not have to include Hook in the name. It is nice and extensible, if we added another form library then we could call it FormikForm etc but in reality I imagine we will only support one form type at any given time so could we just name these `Form` and `FormInput`?